src/averageAges.js

```diff
@@ -55,7 +81,61 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    person['mother']).filter(person => person !== null);
+
+  const mothersArr = people.map(person => person['mother']);
+  const mothers = people.filter(person => mothersArr.includes(person['name']));
+  let countOfpeople = 0;
+  let result = 0;
+
+  if (!onlyWithSon) {
+    const amountOfDifferAges = children.reduce((previousValue, child) => {
+      const mother = mothers.find((mom) => {
+        if (mom['name'] === child['mother']) {
+          return mom;
+        }
+      });
+
+      let sum = previousValue;
+
+      if (mother !== undefined) {
+        sum += child['born'] - mother['born'];
+        countOfpeople++;
+      }
+
+      return sum;
+    }, 0);
+
+    result = amountOfDifferAges / countOfpeople;
+  }
```

Comment:
Your this logic are the same for both of variants: if you have onlyWithSon argument and if not
Move this check at the start of the function as you did for previous functions
Just check at the start, if this argument is passed, you should find all kids that are boys
If this argument isn't passes - you should find all kids
And then logic is the same (reduce and calculating average age)